syntax = "proto3";

package Key;

enum TYPE{
	RANDOMNUM = 0;
	PREKEY = 1;
	KEYCHECK = 2;
	KEYREVOKE = 3;
}

enum RV{
	SUCCESS = 0;
	AUTHERR = 1;
	SERVERERR = 2;
}

message RandomNumRequest
{
	bytes random = 1;
}

message RandomKeyRespone
{
	bytes random = 1;
	bytes key = 2;
}

message PremasterKeyRequest
{
	bytes prekey = 1;
}

message RequestMsg
{
	TYPE type = 1;
    // int32	cmdType = 1;	// 报文类型
	bytes	clientId = 2;	// 客户端编号
	bytes	authCode = 3;	// 认证码
	bytes	serverId = 4;	// 服务器端编号 
	bytes	sign = 5;
	bytes	data = 6;
	int32	keyid = 7;
	// oneof request{
	// 	RandomNumRequest randomNumReq = 6;
	// 	PremasterKeyRequest premasterKeyReq = 7;
	// }
}

message RespondMsg
{
	TYPE type = 1;
    RV	 rv = 2;		    // 返回值
	bytes	clientId = 3;	// 客户端编号
	bytes	serverId = 4;	// 服务器编号
	// bytes	key = 4;		// 公钥
	// int32	seckeyid = 5;	// 对称公钥编号    keysn
	bytes sign = 5;
	bytes authCode = 6;
	bytes data = 7;
	// oneof respond{
	// 	RandomKeyRespone randomKeyRes = 4;
	// }
}